components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -394,10 +399,21 @@ public User createUser(User user, Map<String, Boolean> requiredAttributes)
                 if (isResourceLimitReachedError((UserStoreClientException) ex)) {
                     handleResourceLimitReached();
                 }
+                publishEventOnUserRegistrationFailure(user, ResponseCodeConstants.INVALID_VALUE, ex.getMessage(),
+                        claimsInLocalDialect, carbonUM);
                 throw new BadRequestException(errorMessage, ResponseCodeConstants.INVALID_VALUE);
             }
-            handleErrorsOnUserNameAndPasswordPolicy(e);
+
+            try {
+                handleErrorsOnUserNameAndPasswordPolicy(e);
+            } catch (BadRequestException | CharonException exception) {
+                publishEventOnUserRegistrationFailure(user, exception.getScimType(), exception.getMessage(),
+                        claimsInLocalDialect, carbonUM);
+                throw exception;
+            }
         } catch (NotImplementedException e) {
+            publishEventOnUserRegistrationFailure(user, e.getScimType(),
```

Comment:
I don't think we should send a failure message with regard to something not implemented